sources/packages/backend/libs/services/src/queue/queue.service.ts

```diff
@@ -1,12 +1,14 @@
 import { Injectable } from "@nestjs/common";
 import { InjectRepository } from "@nestjs/typeorm";
 import { QueueConfiguration } from "@sims/sims-db";
-import { QueueConfigurationDetails } from "@sims/sims-db/entities/queue-configuration.type";
-import { FindOptionsSelect, Repository } from "typeorm";
+import { QueueNames } from "@sims/utilities";
+import Bull from "bull";
+import { Repository } from "typeorm";
 import { QueueModel } from "./model/queue.model";
 
 @Injectable()
 export class QueueService {
+  private queueConfiguration: QueueConfiguration[] = undefined;
```

Comment:
As a suggestion, the object can be converted t a `Record<string, QueueConfiguration>` where the string will be the unique queue name. In this way, we can have direct access to the cached item instead of executing a filter every time.